lib/lte_link_control/lte_lc_helpers.c

```diff
@@ -981,6 +981,345 @@ int parse_ncellmeas(const char *at_response, struct lte_lc_cells_info *cells)
+		if (err) {
+			LOG_ERR("Could not parse ta_meas_time, error: %d", err);
+			goto clean_exit;
+		}
+
+		/* <earfcn> */
+		curr_index++;
+		err = at_params_int_get(&resp_list, curr_index,
+					&parsed_cell.earfcn);
+		if (err) {
+			LOG_ERR("Could not parse earfcn, error: %d", err);
+			goto clean_exit;
+		}
+
+		/* <phys_cell_id> */
+		curr_index++;
+		err = at_params_short_get(&resp_list, curr_index,
+					&parsed_cell.phys_cell_id);
+		if (err) {
+			LOG_ERR("Could not parse phys_cell_id, error: %d", err);
+			goto clean_exit;
+		}
+
+		/* <rsrp> */
+		curr_index++;
+		err = at_params_short_get(&resp_list, curr_index,
+					&parsed_cell.rsrp);
+		if (err) {
+			LOG_ERR("Could not parse rsrp, error: %d", err);
+			goto clean_exit;
+		}
+
+		/* <rsrq> */
+		curr_index++;
+		err = at_params_short_get(&resp_list, curr_index,
+					&parsed_cell.rsrq);
+		if (err) {
+			LOG_ERR("Could not parse rsrq, error: %d", err);
+			goto clean_exit;
+		}
+
+		/* <meas_time> */
+		curr_index++;
+		err = at_params_int64_get(&resp_list, curr_index,
+					&parsed_cell.measurement_time);
+		if (err) {
+			LOG_ERR("Could not parse meas_time, error: %d", err);
+			goto clean_exit;
+		}
+
+		/* <serving> */
+		curr_index++;
+		err = at_params_short_get(&resp_list, curr_index, &tmp_short);
+		if (err) {
+			LOG_ERR("Could not parse serving, error: %d", err);
+			goto clean_exit;
+		}
+		parsed_cell.serving_cell = tmp_short;
+
+		/* <neighbor_count> */
+		curr_index++;
+		err = at_params_short_get(&resp_list, curr_index, &tmp_short);
+		if (err) {
+			LOG_ERR("Could not parse neighbor_count, error: %d", err);
+			goto clean_exit;
+		}
+		parsed_cell.ncells_count = tmp_short;
+
+		if (parsed_cell.serving_cell) {
+			int to_be_parsed_ncell_count = 0;
+
+			/* This a current / serving cell.
+			 * In practice the <neighbour_count> is always 0 for other than
+			 * the serving cell, i.e. no neigbour cell list is available.
+			 * Thus, handle neighbor cells only for the serving cell.
+			 */
+			cells->current_cell = parsed_cell;
+			if (parsed_cell.ncells_count != 0) {
+				/* Allocate room for the parsed neighbor info. */
+				if (parsed_cell.ncells_count >
+					CONFIG_LTE_NEIGHBOR_CELLS_MAX) {
+					to_be_parsed_ncell_count = CONFIG_LTE_NEIGHBOR_CELLS_MAX;
+					incomplete = true;
+					LOG_WRN("Cutting response, because received neigbor cell"
+						" count is bigger than configured max: %d",
+							CONFIG_LTE_NEIGHBOR_CELLS_MAX);
+
+				} else {
+					to_be_parsed_ncell_count = parsed_cell.ncells_count;
+				}
+				ncells = k_calloc(
+						to_be_parsed_ncell_count,
+						sizeof(struct lte_lc_ncell));
+				if (ncells == NULL) {
+					LOG_WRN("Failed to allocate memory for the ncells"
+						" (continue)");
+					continue;
+				}
+				cells->neighbor_cells = ncells;
+				cells->ncells_count = to_be_parsed_ncell_count;
+			}
+
+			/* Parse neighbours */
+			for (j = 0; j < to_be_parsed_ncell_count; j++) {
+				/* <n_earfcn>j */
```

Comment:
Also here the index should be part of the "label". Maybe this could be for example`<n_earfcn[j]>`? You could use the same format also in the comment in the beginning of function `parse_ncellmeas_gci()`.

Reply:
Changed as proposed.